### PR TITLE
KAFKA-10095: Add stricter assertion in LogCleanerManagerTest

### PR DESCRIPTION
**Pull request by divijvaidya**
### Changes:

Add a stricter assertion in LogCleanerManagerTest
Minor cosmetic simplification in Scala

### Testing:
LogCleanerManagerTest run is successful. All unit test runs are successful using ./gradlew unitTest

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by divijvaidya** [comment]
Requesting review from @abbccdda and @hachikuji since you folks were part of a previous PR for this JIRA.

Please take a look when you get a chance 🙏

File: core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala [hunk 1]
```diff
@@ -413,7 +413,7 @@ class LogCleanerManagerTest extends Logging {
 
     // updateCheckpoints should remove the topicPartition data in the logDir
     cleanerManager.updateCheckpoints(logDir, partitionToRemove = Option(topicPartition))
-    assertTrue(cleanerManager.allCleanerCheckpoints.get(topicPartition).isEmpty)
+    assertTrue(!cleanerManager.allCleanerCheckpoints.contains(topicPartition))
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala]: Would `assertFalse(cleanerManager.allCleanerCheckpoints.contains(topicPartition))` be even more readable?

File: core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala [hunk 2]
```diff
@@ -710,14 +710,16 @@ class LogCleanerManagerTest extends Logging {
...
-    cleanerManager.doneCleaning(topicPartition, log.dir, 1)
+    val endOffset = 1L
+    cleanerManager.doneCleaning(topicPartition, log.dir, endOffset)
     assertTrue(cleanerManager.cleaningState(topicPartition).isEmpty)
-    assertTrue(cleanerManager.allCleanerCheckpoints.get(topicPartition).nonEmpty)
+    assertTrue(cleanerManager.allCleanerCheckpoints.contains(topicPartition))
+    assertEquals(Some(endOffset), cleanerManager.allCleanerCheckpoints.get(topicPartition))
 
     cleanerManager.setCleaningState(topicPartition, LogCleaningAborted)
     cleanerManager.doneCleaning(topicPartition, log.dir, 1)
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala]: Should we reuse `endOffsets` here too?

**Review by mimaison — COMMENTED** [review]
Thanks for the PR. I've left a couple of suggestions

File: core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala (hunk 1 above)
Comment (divijvaidya) [reply core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala]: done.

File: core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala (hunk 2 above)
Comment (divijvaidya) [reply core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala]: done

File: core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala (hunk 1 above)
Comment (mimaison) [reply core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala]: There's a few more `assertTrue(!` below, can we update them too?

**Review by mimaison — APPROVED** [review]
Thanks for the quick update! LGTM

**Comment by mimaison** [comment]
Tests passed locally, merging

**Comment by divijvaidya** [comment]
Thank you for supporting this PR with the review @mimaison. Appreciate it! This was my first contribution to Apache Kafka 🥳

**Comment by mimaison** [comment]
Congratulations @divijvaidya ! Welcome to the community, I'm looking forward to seeing your next contributions

**Pull request merged**